php/get-pkce-access-token

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env php
+<?php
+require "vendor/autoload.php";
+use GuzzleHttp\Client;
+
+const AUTHORIZE_URL = "https://auth.aweber.com/oauth2/authorize";
+const TOKEN_URL = "https://auth.aweber.com/oauth2/token";
+
+
+echo 'Do you wish to create(c) tokens or refresh(r) tokens? ';
+$createRefresh = rtrim(fgets(STDIN), PHP_EOL);
+
+if (strtoupper($createRefresh) == 'C') {
+    // Collect the client ID and set the out of band redirect.
+    // Alternatively, for a browser based flow you could ask for the 
+    // redirect URI here instead, or enter your own.
+    echo "Enter your client ID: ";
+    $clientId = rtrim(fgets(STDIN), PHP_EOL);
+    $redirectUri = "urn:ietf:wg:oauth:2.0:oob";
+
+    // Generate the code challenge using the OS / cryptographic random function
+    $verifierBytes = random_bytes(64);
+    $codeVerifier = rtrim(strtr(base64_encode($verifierBytes), "+/", "-_"), "=");
+
+    // Very important, "raw_output" must be set to true or the challenge
+    // will not match the verifier.
+    $challengeBytes = hash("sha256", $codeVerifier, true);
```

Comment:
To use the hash extension we should have `"ext-hash": "*"` to the requirements in composer.json

Reply:
Whoops, I forgot to commit composer.json. Added it in my recent commit.